src/main/java/gov/nasa/jpl/mbee/mdk/systems_reasoner/actions/AspectSelectionAction.java

```diff
@@ -51,13 +50,47 @@ public void actionPerformed(ActionEvent e) {
 
         final Frame dialogParent = MDDialogParentProvider.getProvider().getDialogParent();
         final ElementSelectionDlg dlg = ElementSelectionDlgFactory.create(dialogParent);
-        SessionManager.getInstance().createSession("Creating aspect.");
+        SessionManager.getInstance().createSession(Application.getInstance().getProject(), "Creating aspect.");
         final SelectElementTypes set = new SelectElementTypes(types, types, null, null);
         final SelectElementInfo sei = new SelectElementInfo(true, false,
                 Application.getInstance().getProject().getModel().getOwner(), true);
+
+        // Collection<BaseElement> allElements = Application.getInstance().getProject().getAllElements();
+
+        ArrayList<Classifier> aspects = new ArrayList<>();
+        findAspectsOfClassifier(aspects);
+        ElementSelectionDlgFactory.initMultiple(dlg,set,sei,aspects);
+
+        dlg.show();
+
+        // Check if the user has clicked "Ok".
+        if (dlg.isOkClicked())
+        {
+            // Get selected element in single selection mode.
+        //    BaseElement selected = dlg.getSelectedElement();
+
+            // Get selected elements in multiple selection mode.
+            List<BaseElement> selected = dlg.getSelectedElements();
+
+            for(BaseElement be : selected){
+                System.out.println(be.getHumanName());
+                Stereotype aspectSt = Utils.getStereotype("aspect");
```

Comment:
Stereotypes should be gotten by ID using StereotypesHelper. ID should be a static constant.